ghost/core/test/e2e-webhooks/tags.test.js

```diff
@@ -100,4 +100,52 @@ describe('tag.* events', function () {
                 }
             });
     });
+
+    it('tag.edited event is triggered', async function () {
+        const webhookURL = 'https://test-webhook-receiver.com/tag-edited/';
+        await webhookMockReceiver.mock(webhookURL);
+        await fixtureManager.insertWebhook({
+            event: 'tag.edited',
+            url: webhookURL
+        });
+    
+        const res = await adminAPIAgent
+            .post('tags/')
+            .body({
+                tags: [{
+                    name: 'Test Tag 3',
+                    slug: 'test-tag-3',
+                    description: 'Test description'
+                }]
+            })
+            .expectStatus(201);
+    
+        const id = res.body.tags[0].id;
+    
+        const updatedTag = res.body.tags[0];
+        updatedTag.name = 'Updated Tag 3';
+        updatedTag.slug = 'updated-tag-3';
+    
+        await adminAPIAgent
+            .put('tags/' + id)
+            .body({
+                tags: [updatedTag]
+            })
+            .expectStatus(200);
+    
+        await webhookMockReceiver.receivedRequest();
+    
+        webhookMockReceiver
+            .matchHeaderSnapshot({
+                'content-version': anyContentVersion,
+                'content-length': anyNumber,
+                'user-agent': anyGhostAgent
+            })
+            .matchBodySnapshot({
+                tag: {
+                    current: {...tagSnapshot, url: anyLocalURL},
```

Comment:
Imo, worth including the `url: anyLocalURL` bit in the shared `tabSnapshot` object now that it's been used in multiple places. Have you tried moving it @dshubhadeep ?

Reply:
Yes, it was done like that before. But tests were failing for tag.deleted webhook since the `previous` key in the snapshot didn't have the URL.
Maybe I could create a new variable something like snapshotWithUrl ?

Reply:
Lets keep the extraction to the [rule of 3](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)). If we have to reuse the same form 3rd time lets extract a `tagSnapshotWithURL` like you proposed or create a builder function having a "withURL: boolean" parameter. It's been used 2 times now, so it's good to go as is 👍